lib/qr_code/data_masking.ex

```diff
@@ -37,7 +37,7 @@ defmodule QRCode.DataMasking do
 
     masking_matrices
     |> Enum.at(index)
-    |> Result.map(fn matrix -> %{qr | matrix: matrix, mask_num: index} end)
+    |> Result.and_then(fn matrix -> %{qr | matrix: matrix, mask_num: index} end)
```

Comment:
Why is here `and_then`? Supplied lambda function does not return `{:ok, ...}` or `{:error, ...}` value.